ext/openssl/openssl.c

```diff
@@ -6825,75 +6885,80 @@ PHP_FUNCTION(openssl_cipher_iv_length)
 	}
 
-	cipher_type = EVP_get_cipherbyname(method);
-	if (!cipher_type) {
-		php_error_docref(NULL, E_WARNING, "Unknown cipher algorithm");
+	if ((ret = php_openssl_cipher_iv_length(method)) == -1) {
 		RETURN_FALSE;
 	}
 
-	RETURN_LONG(EVP_CIPHER_iv_length(cipher_type));
+	RETURN_LONG(ret);
 }
 /* }}} */
 
 
-/* {{{ proto string openssl_random_pseudo_bytes(int length [, &bool returned_strong_result])
-   Returns a string of the length specified filled with random pseudo bytes */
-PHP_FUNCTION(openssl_random_pseudo_bytes)
+PHP_OPENSSL_API zend_string* php_openssl_random_pseudo_bytes(zend_long buffer_length)
 {
-	zend_long buffer_length;
 	zend_string *buffer = NULL;
-	zval *zstrong_result_returned = NULL;
-
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l|z", &buffer_length, &zstrong_result_returned) == FAILURE) {
-		return;
-	}
-
-	if (zstrong_result_returned) {
-		ZEND_TRY_ASSIGN_FALSE(zstrong_result_returned);
-	}
-
 	if (buffer_length <= 0
 #ifndef PHP_WIN32
 		|| ZEND_LONG_INT_OVFL(buffer_length)
 #endif
 			) {
 		zend_throw_exception(zend_ce_error, "Length must be greater than 0", 0);
-		return;
+		return NULL;
 	}
 	buffer = zend_string_alloc(buffer_length, 0);
 
 #ifdef PHP_WIN32
 	/* random/urandom equivalent on Windows */
-	if (php_win32_get_random_bytes((unsigned char*)buffer->val, (size_t) buffer_length) == FAILURE){
+	if (php_win32_get_random_bytes((unsigned char*)(buffer)->val, (size_t) buffer_length) == FAILURE){
 		zend_string_release_ex(buffer, 0);
-		if (zstrong_result_returned) {
-			ZEND_TRY_ASSIGN_FALSE(zstrong_result_returned);
-		}
+		buffer = 0;
 		zend_throw_exception(zend_ce_exception, "Error reading from source device", 0);
-		return;
+		return NULL;
 	}
 #else
 
-	PHP_OPENSSL_CHECK_LONG_TO_INT(buffer_length, length);
+	PHP_OPENSSL_CHECK_LONG_TO_INT_NORET(buffer_length, length);
 	PHP_OPENSSL_RAND_ADD_TIME();
 	/* FIXME loop if requested size > INT_MAX */
 	if (RAND_bytes((unsigned char*)ZSTR_VAL(buffer), (int)buffer_length) <= 0) {
 		zend_string_release_ex(buffer, 0);
-		if (zstrong_result_returned) {
-			ZEND_TRY_ASSIGN_FALSE(zstrong_result_returned);
-		}
+		buffer = 0;
 		zend_throw_exception(zend_ce_exception, "Error reading from source device", 0);
-		return;
+		return NULL;
 	} else {
 		php_openssl_store_errors();
 	}
 #endif
+	return buffer;
+}
 
-	ZSTR_VAL(buffer)[buffer_length] = 0;
-	RETVAL_NEW_STR(buffer);
+/* {{{ proto string openssl_random_pseudo_bytes(int length [, &bool returned_strong_result])
+   Returns a string of the length specified filled with random pseudo bytes */
+PHP_FUNCTION(openssl_random_pseudo_bytes)
+{
+	zend_string *buffer = NULL;
+	zend_long buffer_length;
+	zval *zstrong_result_returned = NULL;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l|z/", &buffer_length, &zstrong_result_returned) == FAILURE) {
+		return;
+	}
+
+	if (zstrong_result_returned) {
+		ZEND_TRY_ASSIGN_FALSE(zstrong_result_returned);
+		zval_ptr_dtor(zstrong_result_returned);
+		ZVAL_FALSE(zstrong_result_returned);
```

Comment:
Probably something went wrong during a rebase here. This should only have the `ZEND_TRY_ASSIGN_FALSE` line, the other two are not necessary. (Same issue below.)

Reply:
Yes was complicated rebasing :-) makes sense your remarks.